src/library/scala/collection/Iterator.scala

```diff
@@ -1016,6 +1047,12 @@ object Iterator extends IterableFactory[Iterator] {
     if (step == 0) throw new IllegalArgumentException("zero step")
     private[this] var i = start
     private[this] var hasOverflowed = false
+    override def knownSize: Int = {
```

Comment:
> I'm inclined to mistrust floating point computations for stuff like this.

So am I, but I figured this should be relatively safe since we're working with numbers < 2 * Int.MaxValue. But would still be good to avoid it, and avoid the code duplication.

Reply:
I can't tell on sight which of the range Iterators is strictly better, but replacing `new RangeIterator(...)` with `Iterator.range(...)` seems a lot more practical.

Reply:
I haven't looked too closely at the implementation to say which one to prefer but `RangeIterator` is private and `Iterator.range` is even anonymous so it won't have any impact on the API. Users will only create them through some method call anyway.